### ISSUE TITLE
Unknown CMake command "pybind11_add_module"

### ISSUE THREAD

**Issue by lucabaldesi** [bastibl/gr-foo#30]
Hi,
thank you for your amazing software. I am trying to compile gr-foo, on maint-3.9 while having installed gnuradio 3.9.2.
I get the following configuration error:

-- Build type not specified: defaulting to release.
-- UHD library found.
-- Using GMP.
-- User set python executable /usr/bin/python3
-- Extracting version information from git describe...
-- Using install prefix: /usr/local
-- Building for version: cc8bfc6e / 1.1.0git
-- PYTHON and GRC components are enabled
-- Python checking for pygccxml - found
CMake Error at /usr/lib64/cmake/gnuradio/GrPybind.cmake:243 (pybind11_add_module):
  Unknown CMake command "pybind11_add_module".
Call Stack (most recent call first):
  python/bindings/CMakeLists.txt:46 (GR_PYBIND_MAKE_OOT)

I have pybind11 with version 2.7.1-1 and cmake 3.21.3
What should I change?

Best,
Luca

**Comment by bastibl** [comment]
I'm not sure what's going on there. But this seems like an issue with your GNU Radio installation, since the error originates in `/usr/lib64/cmake/gnuradio/GrPybind.cmake`.

This module doesn't do anything fancy at `python/bindings/CMakeLists.txt:46`. AFAIS, other 3.9 OOTs do the same. Can you please check, if any other 3.9 OOT works for your (e.g., `gr-satellites`)?

There was also another PyBind issue. Maybe there's some information in there, even though it doesn't seem directly related:
https://github.com/bastibl/gr-rds/issues/48

**Comment by lucabaldesi** [comment]
Turns out this issue shows up only with gnuradio 3.9.2 and it is fixed with 3.9.3.
Anyway, to fix it with gnuradio 3.9.2, add the following lines in CMakeLists.txt:

```
########################################################################
# PyBind11 Related
########################################################################

find_package(pybind11 REQUIRED)
execute_process(
    COMMAND "${PYTHON_EXECUTABLE}" -c
    "try:\n import numpy\n import os\n inc_path = numpy.get_include()\n if os.path.exists(os.path.join(inc_path, 'numpy', 'arrayobject.h')):\n  print(inc_path, end='')\nexcept:\n pass"
    OUTPUT_VARIABLE PYTHON_NUMPY_INCLUDE_DIR)
```
